security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -331,61 +407,75 @@ func (sf *SecretFetcher) scrtUpdated(oldObj, newObj interface{}) {
 		return
 	}
 
-	oldCert, oldKey, oldRoot, _ := extractCertAndKey(oscrt)
-	newCert, newKey, newRoot, valid := extractCertAndKey(nscrt)
-	if !valid {
-		secretFetcherLog.Warnf("skip updating secret. Kubernetes secret: %s has empty field", newScrtName)
+	t := time.Now()
```

Comment:
can we just remove all existing complicated logic in ```scrtUpdated``` since it's not actually in used ?

Reply:
The secret updating logic is removed as rotating secret does not trigger this callback.
We expect the old secret and new secret passed to scrtUpdated remain unchanged. I add some  warning logs if secret has changed.